### PR TITLE
Json formatting working

### PR DESCRIPTION
**Pull request by DarthVaderInc**

### CODE REVIEW

File: include/octo-logger-cpp/sink.hpp [hunk 1]
```diff
@@ -25,18 +25,56 @@ namespace octo::logger
...
+        case LineFormat::PLAINTEXT_LONG:
+          return formatted_log_plaintext_long(log, channel,  context_info, disable_context_info);
+        case LineFormat::PLAINTEXT_SHORT:
+          return formatted_log_plaintext_short(log, channel);
+#ifdef OCTO_LOGGER_JSON_FORMATTING
+        case LineFormat::JSON:
+          return formatted_log_JSON(log, channel,  context_info);
+#endif
+      }
+      throw std::runtime_error("Should never happen!!!");
```
Comment (ofiriluz) [inline include/octo-logger-cpp/sink.hpp:63]:
```suggestion
      throw std::runtime_error("Unexpected Error Occurred");
```

File: include/octo-logger-cpp/sink.hpp [hunk 2]
```diff
@@ -25,18 +25,56 @@ namespace octo::logger
...
 class Sink
 {
+  public:
+    enum class LineFormat {
+      PLAINTEXT_LONG = 0,    // Used in console
+      PLAINTEXT_SHORT = 1,   // Used in syslog
+#ifdef OCTO_LOGGER_JSON_FORMATTING
+      JSON = 2,
+#endif
+      DEFAULT = PLAINTEXT_LONG
```
[the remark this answers is not in the document]
Comment (DarthVaderInc) [reply include/octo-logger-cpp/sink.hpp:35]: So:
static constexpr LineFormat DEFAULT = LineFormat::PLAINTEXT_LONG?

File: CMakeLists.txt [hunk 3]
```diff
@@ -6,9 +6,16 @@ PROJECT(octo-logger-cpp)
 SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake/)
 INCLUDE(ConfigOptions)
 
+SET (JSON_ENABLED OFF)
```
[the remark this answers is not in the document]
Comment (DarthVaderInc) [reply CMakeLists.txt:9]: This is an internal variable to the cmake.
Notice the next lines:
IF (WITH_AWS OR WITH_JSON_FORMATTING )
    SET (JSON_ENABLED ON)
ENDIF()

so basically:
JSON_ENABLED = WITH_AWS OR WITH_JSON_FORMATTING

It is by definition different than WITH_JSON_FORMATTING

File: include/octo-logger-cpp/sink.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (DarthVaderInc) [reply include/octo-logger-cpp/sink.hpp]: Or maybe none at all?
After all, each sink has its own default.

**Review by ofiriluz — APPROVED** [review]

**Pull request merged**